### PR TITLE
Enable war deployment model (#197.1)

### PR DESCRIPTION
**Pull request by wwelling**
**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues)**: https://github.com/vivo-project/VIVO/issues/3034

See details on previous reverted PR: https://github.com/vivo-project/VIVO/pull/197

# What's new?

- Moved `Vitro.defaultNamespace` and `rootUser.emailAddress` from runtime properties to context.xml.
- Maven build command no longer requires settings.xml
- Maven command no longer deploys artifact to Tomcat
- VIVO home directory provisioned on startup if not exists.

# How should this be tested?

Clone Vitro and VIVO and checkout feature branches `VIVO-3034-psuedo-rebase`
```
git clone --branch VIVO-3034-psuedo-rebase git@github.com:vivo-project/Vitro.git
git clone --branch VIVO-3034-psuedo-rebase git@github.com:vivo-project/VIVO.git
```
or
```
git clone git@github.com:vivo-project/Vitro.git
cd Vitro
checkout VIVO-3034-psuedo-rebase
cd ..
git clone git@github.com:vivo-project/VIVO.git
cd VIVO
checkout VIVO-3034-psuedo-rebase
```

Build Vitro

```
cd Vitro
mvn clean install
```

Build and run VIVO with Docker
```
cd VIVO
mvn clean package
docker-compose up
```

If not using Docker, copy the war file into Tomcat.
```
cp installer/webapp/target/vivo.war /usr/local/tomcat/webapps/ROOT.war
```

Check http://localhost:8080/

# Interested parties
@VIVO-project/vivo-committers @chenejac

### CODE REVIEW

**Review by chenejac — DISMISSED** [review]
@wwelling it works for me. Please, double check two my comments.

File: installer/pom.xml [hunk 1]
```diff
@@ -248,7 +198,7 @@
         <repository>
             <id>vivo-dependencies</id>
             <name>VIVO Dependencies</name>
-            <url>https://raw.github.com/vivo-project/dependencies/main/</url>
+            <url>https://raw.github.com/vivo-project/dependencies/master/</url>
```
[the remark this answers is not in the document]
Comment (wwelling) [reply installer/pom.xml]: Updated.

File: installer/webapp/src/main/webResources/WEB-INF/classes/log4j.properties [hunk 2]
```diff
@@ -24,7 +24,7 @@
 # debug.log4j.properties exists will be used instead, if it exists, but is not stored in Subversion.
 
 log4j.appender.AllAppender=org.apache.log4j.RollingFileAppender
-log4j.appender.AllAppender.File=${catalina.base}/logs/${app-name}.all.log
+log4j.appender.AllAppender.File= ${catalina.base}/logs/vivo.all.log
```
[the remark this answers is not in the document]
Comment (wwelling) [reply installer/webapp/src/main/webResources/WEB-INF/classes/log4j.properties]: Trying this now.

[the remark this answers is not in the document]
Comment (wwelling) [reply installer/webapp/src/main/webResources/WEB-INF/classes/log4j.properties]: This works using the `app-name` variable.

**Review by kaladay — CHANGES_REQUESTED** [review]
The runtime properties has some commented out code still using `/usr/local`.

See:
```
home/src/main/resources/config/default.runtime.properties:176:#OpenSocial.tokenKeyFile = /usr/local/vivo/data/shindig/openssl/securitytokenkey.txt
home/src/main/resources/config/default.runtime.properties:361:# harvester.location = /usr/local/vivo/harvester/
```

I think these should be updated to `/opt/vivo` paths.

**Comment by wwelling** [comment]
@kaladay these are changing back with the following PR. https://github.com/vivo-project/VIVO/pull/3844 Perhaps the following PR should be reviewed in replacement of this.

**Comment by chenejac** [comment]
superseded by https://github.com/vivo-project/VIVO/pull/3844

**Pull request closed without merging**